src/server/dom.ts

```diff
@@ -1066,18 +1067,22 @@ export function getAttributeTask(selector: SelectorInfo, name: string): Schedula
   }, { parsed: selector.parsed, strict: selector.strict, name });
 }
 
-export function inputValueTask(selector: SelectorInfo): SchedulableTask<string> {
-  return injectedScript => injectedScript.evaluateHandle((injected, { parsed, strict }) => {
+export function inputValueTask(selector: SelectorInfo, expected: channels.ExpectedTextValue | undefined): SchedulableTask<string> {
```

Comment:
I do not see where we parse the SerializedValue inside expected...

Reply:
Good point, I was hoping we can save on double serialization between protocol and injected script, but now I think it isn't a great idea.